### PR TITLE
Improvement TiDB integration

### PR DESCRIPTION
**Pull request by SabaPing**
### What does this PR do?

Improve TiDB integration user experience and bump version.

### Motivation

- Refactor user doc.
- Change `spec.yaml` to expose customized metrics config fields.
- Remove useless monitors.
- Bump version to `1.0.0`.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

File: tidb/CHANGELOG.md [hunk 1]
```diff
@@ -1,2 +1,10 @@
-# CHANGELOG - tidb
+# CHANGELOG - TiDB
 
+## 1.0.0 / 2021-01-25
+
+* [Improvement] Make user doc more understandable.
+* [Removed] Removed useless the monitor.
```
Comment (sarina-dd) [inline tidb/CHANGELOG.md:6]:
```suggestion
* [Removed] Removed useless monitor.
```

Comment (sarina-dd) [reply tidb/CHANGELOG.md:6]: Does this just mean we removed a monitor that is unused, or is there another word missing from this sentence?

File: tidb/README.md [hunk 2]
```diff
@@ -2,17 +2,57 @@
 
 ## Overview
 
-TiDB check monitors the overall health and performance of a [TiDB][1] cluster.
+Connect [TiDB][1] Cluster to Datadog in order to:
```
Comment (sarina-dd) [inline tidb/README.md:5]:
```suggestion
Connect [TiDB][1] cluster to Datadog in order to:
```

File: tidb/README.md [hunk 3]
```diff
@@ -2,17 +2,57 @@
...
+- Visualize cluster performance on the provided dashboard.
+
+> Note: TiDB 4.0+ is required for this integration. Integration of TiDB Cloud with Datadog is not available now.
 
 ## Setup
 
 ### Installation
 
-Manually install the TiDB check:
+First, [Download and launch the Datadog Agent][8].
```
Comment (sarina-dd) [inline tidb/README.md:17]:
```suggestion
First, [download and launch the Datadog Agent][8].
```

File: tidb/README.md [hunk 4]
```diff
@@ -2,17 +2,57 @@
...
+
+> Current TiDB integration version: `1.0.0`
+
+#### Host
+
+Run `datadog-agent integration install -t datadog-tidb==<INTEGRATION_VERSION>`.
+
+#### Containerized
+
+The best way to use an integration from integrations-extra with the Docker Agent is to build the Agent with this integration installed. Use the following Dockerfile to build an updated version of the Agent:
```
Comment (sarina-dd) [inline tidb/README.md:29]:
```suggestion
The best way to use this integration with the Docker Agent is to build the Agent with this integration installed. Use the following Dockerfile to build an updated version of the Agent:
```

File: tidb/README.md [hunk 5]
```diff
@@ -2,17 +2,57 @@
...
+FROM gcr.io/datadoghq/agent:latest
+
+ARG INTEGRATION_VERSION=1.0.0
 
-1. [Download the Datadog Agent](https://app.datadoghq.com/account/settings#agent).
+RUN agent integration install -r -t datadog-tidb==${INTEGRATION_VERSION}
+```
 
-2. Upload the build artifact to any host with an Agent and run `datadog-agent integration install -t datadog-tidb==<INTEGRATION_VERSION>`.
+Build the image and push it to your private docker registry.
```
Comment (sarina-dd) [inline tidb/README.md:39]:
```suggestion
Build the image and push it to your private Docker registry.
```

File: tidb/README.md [hunk 6]
```diff
@@ -2,17 +2,57 @@
...
+
+Then, upgrade the Datadog agent container image. If the Helm chart is used, modify the `agents.image` section in the `values.yaml` to replace the default agent image:
```
Comment (sarina-dd) [inline tidb/README.md:41]:
```suggestion
Then, upgrade the Datadog Agent container image. If the Helm chart is used, modify the `agents.image` section in the `values.yaml` to replace the default agent image:
```

File: tidb/README.md [hunk 7]
```diff
@@ -2,17 +2,57 @@
...
+
+```yaml
+agents:
+  enabled: true
+  image:
+    tag: <NEW_TAG>
+    repository: <YOUR_PRIVATE_REPOSITORY>/<AGENT_NAME>
+```
+
+Use the new `values.yaml` to upgrade agent:
```
Comment (sarina-dd) [inline tidb/README.md:51]:
```suggestion
Use the new `values.yaml` to upgrade the Agent:
```

File: tidb/README.md [hunk 8]
```diff
@@ -85,49 +113,49 @@ _Available for Agent versions >6.0_
...
-            name: new_log_start_with_datetime
-            pattern: \[\d{4}/\d{2}/\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}\s\+\d{2}:\d{2}\]
    ```
 
-   Change the `path` and `service` parameter values and configure them for your environment. See the [sample tidb.d/conf.yaml][3] for all available configuration options.
-
-   To make sure that stacktraces are properly aggregated as one single log, a [multiline processing rule][7] can be added.
+   Change the `path` and `service` according to your cluster's configuration. 
+   
+   You can use these commands to show all log path:
```
Comment (sarina-dd) [inline tidb/README.md:120]:
```suggestion
   Use these commands to show all log path:
```

File: tidb/README.md [hunk 9]
```diff
@@ -85,49 +113,49 @@ _Available for Agent versions >6.0_
...
+   ```
 
+3. [Restart the Agent][4].
 
 #### Containerized
 
 ##### Metric collection
 
-For containerized environments, see the [Autodiscovery Integration Templates][2] for guidance on applying the parameters below.
+For containerized environments, after the installation of TiDB integration to Datadog agent image, auto-discovery is configured.
```
Comment (sarina-dd) [inline tidb/README.md:135]:
```suggestion
For containerized environments, after the installation of TiDB integration to Datadog Agent image, Autodiscovery is configured.
```

File: tidb/README.md [hunk 10]
```diff
@@ -85,49 +113,49 @@ _Available for Agent versions >6.0_
...
+
+Thus, metrics is automatically collected to Datadog server.
```
Comment (sarina-dd) [inline tidb/README.md:137]:
```suggestion
Thus, metrics is automatically collected to Datadog's server.
```

File: tidb/README.md [hunk 11]
```diff
@@ -85,49 +113,49 @@ _Available for Agent versions >6.0_
...
 
-| Parameter            | Value                                                                                                                                                        |
-| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| `<INTEGRATION_NAME>` | `tidb`                                                                                                                                                       |
-| `<INIT_CONFIG>`      | blank or `{}`                                                                                                                                                |
-| `<INSTANCE_CONFIG>`  | `{"pd_metric_url": "http://%%host%%:2379/metrics", "tidb_metric_url": "http://%%host%%:10080/metrics", "tikv_metric_url": "http://%%host%%:20180/metrics}"}` |
+If you need to override default auto-discovery behavior, add Datadog annotations to TiDB Pods:
```
Comment (sarina-dd) [inline tidb/README.md:139]:
```suggestion
If you need to override default Autodiscovery behavior, add Datadog annotations to TiDB Pods:
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 12]
```diff
@@ -25,46 +25,115 @@ instances:
     #
   - pd_metric_url: http://%%host%%:2379/metrics
 
+    ## @param pd_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint in addition to default PD metrcis, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:29]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint in addition to default PD metrics, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 13]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    #   - memory:mem
+    #   - process_resident_memory_bytes: go_runtime.process_resident_memory.bytes
+
     ## @param tidb_metric_url - string - required
     ## The URL where TiDB metrics are exposed by Prometheus.
     #
     tidb_metric_url: http://%%host%%:10080/metrics
 
+    ## @param tidb_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint in addition to default TiDB metrcis, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:43]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint in addition to default TiDB metrics, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 14]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    #   - memory:mem
+    #   - process_resident_memory_bytes: go_runtime.process_resident_memory.bytes
+
     ## @param tikv_metric_url - string - required
     ## The URL where TiKV metrics are exposed by Prometheus.
     #
     tikv_metric_url: http://%%host%%:20180/metrics
 
+    ## @param tikv_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint in addition to default TiKV metrcis, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:57]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint in addition to default TiKV metrics, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 15]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    #   - memory:mem
+    #   - process_resident_memory_bytes: go_runtime.process_resident_memory.bytes
+
     ## @param tiflash_proxy_metric_url - string - optional - default: http://%%host%%:20292/metrics
     ## The URL where TiFlash Proxy metrics are exposed by Prometheus.
     #
     # tiflash_proxy_metric_url: http://%%host%%:20292/metrics
 
+    ## @param tiflash_proxy_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:71]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 16]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    # tiflash_proxy_customized_metrics:
+    #   - tiflash*
+
     ## @param tiflash_metric_url - string - optional - default: http://%%host%%:8234/metrics
     ## The URL where TiFlash metrics are exposed by Prometheus.
     #
     # tiflash_metric_url: http://%%host%%:8234/metrics
 
+    ## @param tiflash_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:83]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 17]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    # tiflash_customized_metrics:
+    #   - tiflash*
+
     ## @param ticdc_metric_url - string - optional - default: http://%%host%%:8301/metrics
     ## The URL where TiCDC Proxy metrics are exposed by Prometheus.
     #
     # ticdc_metric_url: http://%%host%%:8301/metrics
 
+    ## @param ticdc_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:95]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 18]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    # ticdc_customized_metrics:
+    #   - ticdc*
+
     ## @param dm_master_metric_url - string - optional - default: http://%%host%%:8261/metrics
     ## The URL where DM master metrics are exposed by Prometheus.
     #
     # dm_master_metric_url: http://%%host%%:8261/metrics
 
+    ## @param dm_master_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:107]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 19]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    # dm_master_customized_metrics:
+    #   - dm*
+
     ## @param dm_worker_metric_url - string - optional - default: http://%%host%%:8262/metrics
     ## The URL where DM worker metrics are exposed by Prometheus.
     #
     # dm_worker_metric_url: http://%%host%%:8262/metrics
 
+    ## @param dm_worker_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:119]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/auto_conf.yaml [hunk 20]
```diff
@@ -25,46 +25,115 @@ instances:
...
+    # dm_worker_customized_metrics:
+    #   - dm*
+
     ## @param pump_metric_url - string - optional - default: http://%%host%%:8250/metrics
     ## The URL where Pump metrics are exposed by Prometheus.
     #
     # pump_metric_url: http://%%host%%:8250/metrics
 
+    ## @param pump_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint, if there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/auto_conf.yaml:131]:
```suggestion
    ## List of metrics to be fetched from the Prometheus endpoint, if there's a
```

File: tidb/datadog_checks/tidb/data/conf.yaml.example [hunk 21]
```diff
@@ -50,46 +50,115 @@ instances:
...
+    #   - memory:mem
+    #   - process_resident_memory_bytes: go_runtime.process_resident_memory.bytes
+
     ## @param tidb_metric_url - string - required
     ## The URL where TiDB metrics are exposed by Prometheus.
     #
     tidb_metric_url: http://localhost:10080/metrics
 
+    ## @param tidb_customized_metrics - list of strings - optional
+    ## List of metrics to be fetched from the prometheus endpoint in addition to default TiDB metrcis. If there's a
```
Comment (sarina-dd) [inline tidb/datadog_checks/tidb/data/conf.yaml.example:68]: Two things for this file's notes:

- Fix `metrcis` misspellings.
- Capitalize Prometheus.

**Review by sarina-dd — COMMENTED** [review]
Docs review completed

File: tidb/README.md (hunk 10 above)
Comment (CharLotteiu) [reply tidb/README.md:137]:
```suggestion
Thus, metrics are automatically collected to Datadog's server.
```

File: tidb/README.md (hunk 11 above)
Comment (CharLotteiu) [reply tidb/README.md:139]:
```suggestion
If you need to override the default Autodiscovery behavior, add Datadog annotations to TiDB Pods:
```

File: tidb/README.md (hunk 9 above)
Comment (CharLotteiu) [reply tidb/README.md:135]:
```suggestion
For containerized environments, after the TiDB check is integrated in the Datadog Agent image, Autodiscovery is configured by default.
```

**Comment by SabaPing** [comment]
Thanks for reviewing the document.
We plan to add a fine-grained one-screen dashboard in the next week. 

Once this is done, could you please add TiDB Integration to integration-core so that our users don't have to install it manually? (currently the installation process is a bit tedious😣)

**Comment by hithwen** [comment]
Hallo @SabaPing, integrations-core is only for integrations built and maintained by datadog.

**Review by hithwen — APPROVED** [review]

**Pull request merged**